### PR TITLE
[Merged by Bors] - feat: Generalise sign of power lemmas

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

From #11986.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Parity.lean [hunk 1]
```diff
@@ -510,48 +526,79 @@ end Ring
 
 section Powers
 
-set_option linter.deprecated false
+section LinearOrderedSemiring_ExistsAddOfLE
```
Comment (eric-wieser) [inline Mathlib/Algebra/Parity.lean:529]: @YaelDillies, can you add a comment to this section explaining why this is more interesting than `CanonicallyOrderedCommSemiring`?

File: Mathlib/Algebra/Parity.lean [hunk 2]
```diff
@@ -510,48 +526,79 @@ end Ring
...
+  rwa [add_rotate', ← hbd, add_zero, add_left_comm, ← add_assoc, ← hac, zero_add]
+#align odd.strict_mono_pow Odd.strictMono_pow
+
+end LinearOrderedSemiring_ExistsAddOfLE
+
+section LinearOrderedRing
+
+variable [LinearOrderedRing R] {a : R} {n : ℕ}
+
+set_option linter.deprecated false
```
Comment (eric-wieser) [inline Mathlib/Algebra/Parity.lean:600]: Can you make this a `set_option ... in` on the lemmas it is needed on?

**Review by eric-wieser — APPROVED** [review]
bors d=@YaelDillies

File: Mathlib/Algebra/Parity.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Parity.lean:529]: You mean `CanonicallyLinearOrderedCommSemiring` (which doesn't exist)?

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**